qualtran/bloqs/rotations/phase_gradient.py

```diff
@@ -224,11 +269,19 @@ def with_registers(self, *new_registers: Union[int, Sequence[int]]):
     def phase_dtype(self) -> QFxp:
         return QFxp(self.phase_bitsize, self.phase_bitsize, signed=False)
 
+    @cached_property
+    def inp_dtype(self) -> QFxp:
+        return QFxp(self.x_bitsize, self.x_bitsize, signed=False)
+
     @cached_property
     def gamma_dtype(self) -> QFxp:
         n_int = Fxp(abs(self.gamma), signed=False).n_int
         return QFxp(n_int + self.gamma_bitsize, self.gamma_bitsize, signed=False)
 
+    @cached_property
```

Comment:
Why do we take abs(gamma) and not just keep it signed?

Reply:
since `gamma` is a classically known constant, we can decide at the time of circuit construction whether we wish to add or subtract the scaled value into the target register. Thus, we don't need to waste one bit of gamma to store the sign information. This is why `AddIntoPhaseGrad` accepts a `sign: int` which is essentially the `sign(gamma)`